src/client/java/net/neoforged/neoforge/client/resources/NeoSplashHooks.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.resources;
+
+import static java.time.temporal.ChronoField.DAY_OF_MONTH;
+import static java.time.temporal.ChronoField.MONTH_OF_YEAR;
+
+import com.google.common.collect.ListMultimap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.MultimapBuilder;
+import com.mojang.logging.LogUtils;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.DataResult;
+import com.mojang.serialization.JsonOps;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import java.io.IOException;
+import java.io.Reader;
+import java.time.Month;
+import java.time.MonthDay;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.util.Collections;
+import java.util.List;
+
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.packs.resources.Resource;
+import net.minecraft.server.packs.resources.ResourceManager;
+import net.minecraft.util.GsonHelper;
+import net.minecraft.util.profiling.ProfilerFiller;
+import org.jetbrains.annotations.Nullable;
+
+public class NeoSplashHooks {
+    private static final String SPLASHES_LOCATION = "texts/splashes.json";
+    private static final ListMultimap<MonthDay, String> SPECIAL_SPLASHES = MultimapBuilder.hashKeys().arrayListValues().build();
+
+    public static List<String> prepare(List<String> vanilla, ResourceManager resourceManager, ProfilerFiller profilerFiller) {
+        SPECIAL_SPLASHES.clear();
+
+        // Vanilla splashes, including the special case "<capitalized username> IS YOU"
+        List<String> list = Lists.newArrayList(vanilla);
+        list.add("%u IS YOU");
+
+        // Special vanilla splashes
+        SPECIAL_SPLASHES.put(MonthDay.of(Month.DECEMBER, 24), "Merry X-mas!");
+        SPECIAL_SPLASHES.put(MonthDay.of(Month.JANUARY, 1), "Happy new year!");
+        SPECIAL_SPLASHES.put(MonthDay.of(Month.OCTOBER, 31), "OOoooOOOoooo! Spooky!");
```

Comment:
Wouldnt it be better for these be included via some splashes json under the vanilla (`minecraft`) namespace, rather than being hardcoded here?